src/models/loan/paymentCalculations.js

```diff
@@ -0,0 +1,46 @@
+function daysInYear(year) {
+  return 365
+}
+
+function daysInMonth(month) {
+  return 30.417
+}
+
+function effectiveMonthlyInterestRate(yearlyInterestRate, month, year) {
+  return ((1 + yearlyInterestRate / daysInYear(year)) ** daysInMonth(month)) - 1
+}
+
+const calculateRepaymentTerm = ({debt:{amountOwed, interestRate: {rate}}, paymentPlan: {monthlyPayment}}) => {
+  let repaymentTerm
+  const interestRate = effectiveMonthlyInterestRate(rate)
+  if (amountOwed == 0 ) {
+    repaymentTerm = 0
+  } else if (rate) {
+    if (monthlyPayment <= amountOwed * interestRate && amountOwed > 0) {
+      repaymentTerm = Infinity
+    } else {
+      repaymentTerm = (Math.log(monthlyPayment)
+                        - Math.log(monthlyPayment - amountOwed * effectiveMonthlyInterestRate(rate))
+                      ) / Math.log(1 + effectiveMonthlyInterestRate(rate))
+    }
+  } else {
+    repaymentTerm = amountOwed / monthlyPayment
+  }
+
+  return Math.ceil(repaymentTerm)
```

Comment:
out of curiosity, what happens if you call `Math.ceil` on `Infinity`?

Reply:
Infinity